src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/RazorFormattingPass.cs

```diff
@@ -430,7 +430,8 @@ private bool FormatBlock(FormattingContext context, RazorSourceDocument source,
             didFormat = true;
         }
 
-        if (closeBraceNode.TryGetLinePositionSpanWithoutWhitespace(source, out var closeBraceRange) &&
+        if (closeBraceNode.Span.Length > 0 &&
```

Comment:
Without this line, the test on the syntactically invalid document failed, because the compiler apparently adds an empty node at position 0 for the close brace on an invalid document.